src/Get.tsx

```diff
@@ -96,28 +98,37 @@ export interface GetComponentState<T> {
  * debugging.
  */
 class ContextlessGet<T> extends React.Component<GetComponentProps<T>, Readonly<GetComponentState<T>>> {
-  private shouldFetchImmediately = () => !this.props.wait && !this.props.lazy;
-
-  readonly state: Readonly<GetComponentState<T>> = {
+  public readonly state: Readonly<GetComponentState<T>> = {
     data: null, // Means we don't _yet_ have data.
     response: null,
     error: "",
-    loading: this.shouldFetchImmediately(),
+    loading: false,
   };
 
-  componentDidMount() {
-    this.shouldFetchImmediately() && this.fetch();
+  public static getDerivedStateFromProps(props: Pick<GetComponentProps, "wait" | "lazy">) {
+    return { loading: !props.lazy };
+  }
+
+  public componentDidMount() {
+    if (!this.props.lazy) {
+      this.fetch();
+    }
   }
 
-  componentDidUpdate(prevProps: GetComponentProps<T>) {
+  public componentDidUpdate(prevProps: GetComponentProps<T>) {
     // If the path or base prop changes, refetch!
     const { path, base } = this.props;
     if (prevProps.path !== path || prevProps.base !== base) {
-      this.shouldFetchImmediately() && this.fetch();
+      if (!this.props.lazy) {
```

Comment:
Why not use `this.state.loading` here?

Reply:
Because the logic flow here is simply: 

- did something change? 🤔 
- does the user want to lazy load? 🤔 
- DONT DO ANYTHING.

It doesn't quite depend on loading state.

Reply:
Sure, but if I'm in `loading`, that means that I have a request pending, so 2 (or more) requests can arrive in this case -> carefull to concurrency 💥

Reply:
Let's tackle this in a separate PR. I made an issue (#12) where we can handle this.